syndeo-template/src/test/java/com/google/cloud/syndeo/it/KafkaToBigQueryIT.java

```diff
@@ -0,0 +1,291 @@
+  private static final String REGION = TestProperties.region();
+  private static final String TEMP_LOCATION = TestProperties.artifactBucket();
+  private static final Credentials CREDENTIALS = TestProperties.googleCredentials();
+  private static final String KAFKA_BOOTSTRAP_SERVER =
+      System.getProperty("kafka_bootstrap_server", "kafka-pabloem-sept-2022-test-m:9092");
+  private static final String KAFKA_TOPIC = "quickstart-events";
+  private static final Integer KAFKA_TOPIC_PARTITIONS = 27;
+  // The syndeo_test dataset exists in the cloud-teleport-testing project.
+  private static final String BIGQUERY_DATASET =
+      "syndeo_test" + UUID.randomUUID().toString().replaceAll("-", "");
+  private static final String BIGQUERY_TABLE = BIGQUERY_DATASET + ".table" + UUID.randomUUID();
+  private static final String SPEC_PATH =
+      TestProperties.specPath() != null
+          ? TestProperties.specPath()
+          : "gs://cloud-teleport-testing-df-staging/syndeo-template.json";
+
+  private static final Integer ELEMENTS_PER_SECOND = 1_000_000;
+  private static final Integer PIPELINE_DURATION_SECONDS = 600;
+
+  private static final Long ONE_MINUTE_MILLIS = 60 * 1000L;
+  private static final Long SIX_MINUTES_MILLIS = 6 * ONE_MINUTE_MILLIS;
+
+  private DefaultBigQueryResourceManager bigQueryResourceManager;
+
+  @Before
+  public void setUp() {
+    bigQueryResourceManager =
+        DefaultBigQueryResourceManager.builder("kafka-bq-test", PROJECT).build();
+    bigQueryResourceManager.createDataset(BIGQUERY_DATASET);
+  }
+
+  @After
+  public void tearDown() {
+    bigQueryResourceManager.cleanupAll();
+  }
+
+  @Test
+  public void testOnlyWriteDataToKafka() throws Exception {
+    // Make sure that Kafka Server exists
+    // Start data generation pipeline
+    PipelineResult dataGeneratorPipeline = kickstartDataGeneration();
+    if (!DataflowUtils.waitUntilState(
+        dataGeneratorPipeline, PipelineResult.State.RUNNING, SIX_MINUTES_MILLIS)) {
+      dataGeneratorPipeline.cancel();
+      throw new RuntimeException(
+          "Data generator pipeline did not start by deadline. Cancelling test");
+    }
+  }
+
+  @Test
+  public void testKickOffKafkaToBigQuerySyndeoPipeline() throws Exception {
+    // Make sure that Kafka Server exists
+    // Start data generation pipeline
+    PipelineResult dataGeneratorPipeline = kickstartDataGeneration();
+    if (!DataflowUtils.waitUntilState(
+        dataGeneratorPipeline, PipelineResult.State.RUNNING, SIX_MINUTES_MILLIS)) {
+      dataGeneratorPipeline.cancel();
+      throw new RuntimeException(
+          "Data generator pipeline did not start by deadline. Cancelling test");
+    }
+
+    // Start syndeo pipeline.
+    FlexTemplateClient templateClient =
+        FlexTemplateClient.builder().setCredentials(CREDENTIALS).build();
+    {
+      DataflowClient.JobInfo syndeoPipeline = kickstartSyndeoPipeline();
+      DataflowOperator operator = new DataflowOperator(templateClient);
```

Comment:
We might need one dataflow operator only?

Reply:
great catch thank pranav : )